src/globals.ts

```diff
@@ -5,23 +5,32 @@ import { BasicSpan } from './types/spans/basicSpan';
 import {
   getAutoTagKeys,
   getJSONBase64Size,
-  getMaxRequestSize,
-  getMaxRequestSizeOnError,
+  getMaxSizeForStoredSpansInMemory,
   isLambdaTraced,
   spanHasErrors,
 } from './utils';
 import { GlobalDurationTimer } from './utils/globalDurationTimer';
 import { isString } from '@lumigo/node-core/lib/common';
+import { runOneTimeWrapper } from './utils/functionUtils';
 
 const MAX_TAGS = 50;
 const MAX_TAG_KEY_LEN = 50;
 const MAX_TAG_VALUE_LEN = 70;
 const ADD_TAG_ERROR_MSG_PREFIX = 'Skipping addExecutionTag: Unable to add tag';
 export const DEFAULT_MAX_SIZE_FOR_REQUEST = 1024 * 500;
 export const DEFAULT_MAX_SIZE_FOR_REQUEST_ON_ERROR = 1024 * 990;
+export const DEFAULT_MAX_SIZE_FOR_SPANS_SIZE_STORED_IN_MEMORY =
+  DEFAULT_MAX_SIZE_FOR_REQUEST_ON_ERROR * 10;
 export const MAX_TRACER_ADDED_DURATION_ALLOWED = 750;
 export const MIN_TRACER_ADDED_DURATION_ALLOWED = 200;
 
+const warnSpansSizeOnce = runOneTimeWrapper((threshold: number, currentSize: number) => {
+  logger.info(
+    `Lumigo tracer is no longer collecting data on the invocation, because it reached the maximum ` +
+      `size of calls collected of ${threshold} bytes (current size is ${currentSize} bytes)`
```

Comment:
I would suggest:
```suggestion
    `Lumigo tracer is no longer collecting data on the invocation - maximum size of total spans collected (${threshold} bytes allowed, current size is ${currentSize} bytes)`
```
Also, maybe it's better to show KB rather than bytes?

Reply:
I don't know if we should start manipulating the units, especially if the size limitation that the user specifies is always in bytes.

If the user could specify `10 MB` in the env var value then yes I would agree, but otherwise I think we should always refer to the size as bytes to avoid complications